File: Maratona_SBC_de_Programacao/hiper/hiper.py
# Importing the required library
import sys

def main():
    # Reading the number of elements
    n = int(input())
    vet = []
    
    # Reading the elements into the list
    for i in range(n):
        vet.append(int(input()))
    
    # Checking for the condition and printing the result
    for i in range(n):
        cont = 0
        for k in range(2, 11):
            if vet[i] % k == 0 and k != vet[i]:
                print("0")
                cont += 1
                break
        if cont == 0:
            print("1")

if __name__ == "__main__":
    main()

File: Maratona_SBC_de_Programacao/hiper/hiper.c
#include <stdio.h>

int main() {
    int n = 0;
    int cont = 0;
    scanf("%d", &n);
    int vet[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &vet[i]);
    }
    for (int i = 0; i < n; i++) {
        cont = 0;
        for(int k = 2; k <= 10; k++) {
            if (vet[i] % k == 0 && k != vet[i]) {
                printf("0 \n");
                cont++;
                break;
            }
        }
        if(cont == 0) {
            printf("1 \n");
        }
    }

    return 0;
}